codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +277,13 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                int http1Index =
+                    ByteBufUtil.indexOf(HTTP_1_X_BUF, in.slice(in.readerIndex(), min(in.readableBytes(), 1024)));
```

Comment:
I am still not sure where this 1024 comes from. Can you add a comment why 1024 was chosen ?